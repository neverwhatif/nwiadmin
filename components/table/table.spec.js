import React from 'react';
import { shallow } from 'enzyme';

import Table from './index';

jest.mock('./tablehead', () => 'TableHead');
jest.mock('./tablerow', () => 'TableRow');

const data = [{
    $id: 1,
    Title: 'Lorem',
}, {
    $id: 2,
    Title: 'Ipsum',
}];

const mockTransformer = item => ({
    $id: item.$id * 2,
});

describe('Table', () => {
    const component = shallow(<Table data={data} />);
    const componentIsDisabled = shallow(<Table data={data} isDisabled />);
    const componentWithNoData = shallow(<Table data={[]} />);
    const componentWithTransformer = shallow(<Table data={data} transformer={mockTransformer} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render head', () => {
        expect(component.find('TableHead').length).toBe(1);
    });
    it('should render rows', () => {
        expect(component.find('TableRow').length).toBe(2);
    });
    it('should render disabled', () => {
        expect(componentIsDisabled.hasClass('rootDisabled')).toBe(true);
    });
    it('should render nothing if no data', () => {
        expect(componentWithNoData.type()).toBeNull();
    });
    it('should have a default transformer', () => {
        expect(Table.defaultProps.transformer('lorem')).toBe('lorem');
    });
    it('should have a transformer', () => {
        expect(componentWithTransformer.find('TableRow').first().key()).toBe('2');
    });
});
