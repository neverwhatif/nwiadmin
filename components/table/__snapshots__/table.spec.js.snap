// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table
    columns={Array []}
    data={
      Array [
        Object {
          "$id": 1,
          "Title": "Lorem",
        },
        Object {
          "$id": 2,
          "Title": "Ipsum",
        },
      ]
    }
    functions={null}
    hasHead={true}
    headClass={null}
    isDisabled={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <TableHead
            data={
              Object {
                "$id": 1,
                "Title": "Lorem",
              }
            }
            headClass={null}
            isAllSelected={false}
            toggleSelectAll={[Function]}
          />
        </thead>,
        <tbody>
          <TableRow
            data={
              Object {
                "$id": 1,
                "Title": "Lorem",
              }
            }
            functions={null}
            isOdd={false}
            isSelected={false}
            toggleSelect={[Function]}
          />
          <TableRow
            data={
              Object {
                "$id": 2,
                "Title": "Ipsum",
              }
            }
            functions={null}
            isOdd={true}
            isSelected={false}
            toggleSelect={[Function]}
          />
        </tbody>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TableHead
            data={
              Object {
                "$id": 1,
                "Title": "Lorem",
              }
            }
            headClass={null}
            isAllSelected={false}
            toggleSelectAll={[Function]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "data": Object {
              "$id": 1,
              "Title": "Lorem",
            },
            "headClass": null,
            "isAllSelected": false,
            "toggleSelectAll": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "TableHead",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TableRow
              data={
                Object {
                  "$id": 1,
                  "Title": "Lorem",
                }
              }
              functions={null}
              isOdd={false}
              isSelected={false}
              toggleSelect={[Function]}
            />,
            <TableRow
              data={
                Object {
                  "$id": 2,
                  "Title": "Ipsum",
                }
              }
              functions={null}
              isOdd={true}
              isSelected={false}
              toggleSelect={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "data": Object {
                "$id": 1,
                "Title": "Lorem",
              },
              "functions": null,
              "isOdd": false,
              "isSelected": false,
              "toggleSelect": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "TableRow",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "data": Object {
                "$id": 2,
                "Title": "Ipsum",
              },
              "functions": null,
              "isOdd": true,
              "isSelected": false,
              "toggleSelect": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "TableRow",
          },
        ],
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <TableHead
              data={
                Object {
                  "$id": 1,
                  "Title": "Lorem",
                }
              }
              headClass={null}
              isAllSelected={false}
              toggleSelectAll={[Function]}
            />
          </thead>,
          <tbody>
            <TableRow
              data={
                Object {
                  "$id": 1,
                  "Title": "Lorem",
                }
              }
              functions={null}
              isOdd={false}
              isSelected={false}
              toggleSelect={[Function]}
            />
            <TableRow
              data={
                Object {
                  "$id": 2,
                  "Title": "Ipsum",
                }
              }
              functions={null}
              isOdd={true}
              isSelected={false}
              toggleSelect={[Function]}
            />
          </tbody>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TableHead
              data={
                Object {
                  "$id": 1,
                  "Title": "Lorem",
                }
              }
              headClass={null}
              isAllSelected={false}
              toggleSelectAll={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "data": Object {
                "$id": 1,
                "Title": "Lorem",
              },
              "headClass": null,
              "isAllSelected": false,
              "toggleSelectAll": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "TableHead",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TableRow
                data={
                  Object {
                    "$id": 1,
                    "Title": "Lorem",
                  }
                }
                functions={null}
                isOdd={false}
                isSelected={false}
                toggleSelect={[Function]}
              />,
              <TableRow
                data={
                  Object {
                    "$id": 2,
                    "Title": "Ipsum",
                  }
                }
                functions={null}
                isOdd={true}
                isSelected={false}
                toggleSelect={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "data": Object {
                  "$id": 1,
                  "Title": "Lorem",
                },
                "functions": null,
                "isOdd": false,
                "isSelected": false,
                "toggleSelect": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "TableRow",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "data": Object {
                  "$id": 2,
                  "Title": "Ipsum",
                },
                "functions": null,
                "isOdd": true,
                "isSelected": false,
                "toggleSelect": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "TableRow",
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
